src/Grpc.Net.Client/Internal/UserAgentGenerator.cs

```diff
@@ -103,15 +117,15 @@ private static string GetOS()
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                return "windows";
+                return "windows; ";
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                return "osx";
+                return "osx; ";
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                return "linux";
+                return "linux; ";
             }
             else
```

Comment:
nit: do we have a test where OS is empty string? I think in practice that can happen e.g. when on Xamarin Android, Xamarin iOS etc, so not just a theoretical possibility.

Reply:
Added.